templates/themes/catalog/theme.php

```diff
@@ -407,12 +407,14 @@ private function generateRecentPosts($threads) {
 
             $posts = array();
             foreach ($threads as $post) {
-                if ($board['uri'] !== $post['board']) {
+                if (!isset($board) || $board['uri'] !== $post['board']) {
```

Comment:
This is difficult to understand and is better expressed as

`! (isset($board) && ($board['uri'] === $post['board')]))`

In my opinion. Then it is clear that the isset is just a guard for the next expression.

Reply:
Sounds good. The original thinking was that either this is the first visit to that code (no $board set yet) *or* we're on a post with a different board. It is the equivalent to guarding for $board though, so I will make this change.